Day1/P1.py

```diff
@@ -0,0 +1,154 @@
+    return myindex
+
+print(findchars('hello my name is tram','a'))
+print(findchars('hello my name is tram','my'))
+print(findchars('hello my name is tram','z'))
+print(findchars('hello my name is tram','lo'))
+print(findchars('hello my name is tram','a',12))
+print(findchars('hello my name is tram','m',8,12))
+print(findchars('ohmymymyabcdefmy','my'))
+print('-'*20)
+
+#Convert string into lower case
+txt = 'Hello, And Welcome To My World!'
+print(txt.lower())
+
+def lowerstr(txt):
+    uppertolower = {'A':'a','B':'b','C':'c','D':'d','E':'e','F':'f','G':'g','H':'h','I':'i','J':'j','K':'k','L':'l','M':'m','N':'n','O':'o','P':'p','Q':'q','R':'r','S':'s','T':'t','U':'u','V':'v','W':'w','X':'x','Y':'y','Z':'z'}
+    for i in range(len(txt)):
+        if txt[i] in uppertolower.keys():
+            txt = txt[0:i] + uppertolower[txt[i]] + txt[i+1:]
+    return txt
+
+print(lowerstr('Hello, And Welcome To My World!'))
+print(lowerstr('HSKHNKXjkjskfeKWEOIIHF'))
+print(lowerstr('hello how are you'))
+print('-'*20)
+
+#upper case all characters
+def upperstr(txt):
+    lowertoupper = {'a':'A','b':'B','c':'C','d':'D','e':'E','f':'F','g':'G','h':'H','i':'I','j':'J','k':'K','l':'L','m':'M','n':'N','o':'O','p':'P','q':'Q','r':'R','s':'S','t':'T','u':'U','v':'V','w':'W','x':'X','y':'Y','z':'Z'}
+    for i in range(len(txt)):
+        if txt[i] in lowertoupper.keys():
+            txt = txt[0:i] + lowertoupper[txt[i]] + txt[i+1:]
+    return txt
+
+print(upperstr('Hello, And Welcome To My World!'))
+print(upperstr('HSKHNKXjkjskfeKWEOIIHF'))
+print(upperstr('HELLO HOW ARE YOU'))
+print('-'*20)
+
+#count the number of times a specified value appears in the string
+def countvalue(txt,char,start=None,end=None):
+    count = 0
+    for i in findchars(txt,char,start,end):
+        count += 1
+    return count
+
+print(countvalue('hello my name is tram','a'))
+print(countvalue('hello my name is tram','my'))
+print(countvalue('hello my name is tram','z'))
+print(countvalue('hello my name is tram','lo'))
+print(countvalue('hello my name is tram','a',12))
+print(countvalue('hello my name is tram','m',8,12))
+print(countvalue('ohmymymyabcdefmy','my'))
+print(countvalue('I love apples, apple are my favorite fruit','apple'))
+print('-'*20)
+
+#string capitalize method: change first character in a string to upper case, while other characters remain lower case
+def capitalizestr(txt):
+    alphabet = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
+    if txt[0] in alphabet:
+        txt = upperstr(txt[0]) + txt[1:]
+    return txt
+
+print(capitalizestr('zebras are African equines with distinctive black-and-white striped coats.'))
+print(capitalizestr('All string methods returns new values. They do not change the original string.'))
+print(capitalizestr('HELLO WORLD'))
+print('-'*20)
+
+#isupper() - returns True if all characters in the string are upper case
+def isuppercase(txt):
+    if txt == upperstr(txt):
```

Comment:
This is over-complicated. 

Translating this to words will be: If I want to check whether a string is all upper cases, I will convert a copy of that string to upper case, and then see if that copy is equal to the original string!

How efficient does that sentence sound??? What will you normally do instead?

Using a helper method is to make sure we don't repeat work, it must not make our implementation more complicated!